### PR TITLE
Delete and update Front end guidelines docs

### PR DESCRIPTION
**Pull request by cfarm**
Deleted obsolete docs and updated the atomic structure info with some additional details.

## Additions

- links and examples for atomic components and templates

## Removals

- obsolete `notes-on-atomic-design.md` and `docs/notes-on-frameworks.md` which are older todos/ideas from the v1 project and do not belong as how-tos in our documentation.
- `front-end-testing.md` and `front-end-build.md` since these guides are now in the development repo.
- todos/ideas from the Atomic structure documentation. these don't provide useful information to contributors, they are more suggestions of future improvements that belong in platform repo if we want to revisit them.

## Testing

1. `pip install -r requirements/manual.txt`
1. `mkdocs serve`
1. check out http://127.0.0.1:8000/atomic-structure/

## Notes

-

## Todos

-

## Checklist

* [ ] PR has an informative and human-readable title
* [ ] Changes are limited to a single goal (no scope creep)
* [ ] Code can be automatically merged (no conflicts)
* [ ] Code follows the standards laid out in the [development playbook](https://github.com/cfpb/development)
* [ ] Passes all existing automated tests
* [ ] Any *change* in functionality is tested
* [ ] New functions are documented (with a description, list of inputs, and expected output)
* [ ] Placeholder code is flagged
* [ ] Visually tested in supported browsers and devices
* [ ] Project documentation has been updated
* [ ] Reviewers requested with the [Reviewer tool](https://help.github.com/articles/about-pull-request-reviews/) :arrow_right:

### CODE REVIEW

File: docs/atomic-structure.md [hunk 1]
```diff
@@ -142,7 +142,7 @@ atomicHelpers.instantiateAll( '.o-expandable', Expandable );
 
 ### Templates
 
-Prefixed with “t-” in CSS, JavaScript, and HTML.
+Prefixed with “t-” in CSS, JavaScript, and HTML. [View all available templates](https://github.com/cfpb/cfgov-refresh/tree/master/cfgov/jinja2/v1) that can be extended or reused to create pages.
```
Comment (Scotchester) [inline docs/atomic-structure.md]: This prefixing is not something we actually do. Should we remove the reference to that?

File: docs/atomic-structure.md [hunk 2]
```diff
@@ -192,24 +190,9 @@ cfgov-refresh/test/unit_tests/molecules/
 cfgov-refresh/test/unit_tests/organisms/
 ```
 
-#### Proposed Folder Structure
```
Comment (Scotchester) [inline docs/atomic-structure.md]: This proposal should definitely get captured in a Platform issue or other appropriate location.

File: docs/atomic-structure.md [hunk 3]
```diff
@@ -236,17 +218,17 @@ You can get more information by reading the following:
...
 
-#### Code and Related Pull Requests
+#### JavaScript component examples
 
 [View Unit Test](https://github.com/cfpb/cfgov-refresh/pull/916)
 [Expandable example 1](http://jsfiddle.net/0j9u66h0/9/)
 [Expandable example 2](https://jsfiddle.net/cpsyLy3L/2/)
 
-## Component Build Pipeline
+## Component build pipeline
```
Comment (Scotchester) [inline docs/atomic-structure.md]: There are lots of other headings on this page that could also be sentence-cased.

File: docs/atomic-structure.md [hunk 4]
```diff
@@ -236,17 +218,17 @@ You can get more information by reading the following:
 [A Simple Challenge to Classical Inheritance Fans](https://medium.com/javascript-scene/a-simple-challenge-to-classical-inheritance-fans-e78c2cf5eead#.mtrvhcjiw)
 [Composition over Inheritance (Youtube)](https://www.youtube.com/watch?v=wfMtDGfHWpA)
 
-#### Code and Related Pull Requests
+#### JavaScript component examples
 
 [View Unit Test](https://github.com/cfpb/cfgov-refresh/pull/916)
 [Expandable example 1](http://jsfiddle.net/0j9u66h0/9/)
 [Expandable example 2](https://jsfiddle.net/cpsyLy3L/2/)
```
Comment (Scotchester) [inline docs/atomic-structure.md]: These examples are so old, I'm not sure they're relevant anymore. @sebworks?

File: docs/atomic-structure.md [hunk 5]
```diff
@@ -259,12 +241,11 @@ An explanation for the usage of each bundle is contained in scripts.js.
...
 Routes are used to serve JavaScript bundles to the browser based
 on the requested URL or Wagtail page Media property.
-This happens via code contained in base.html.
+This happens via code contained in [base.html](https://github.com/cfpb/cfgov-refresh/blob/master/cfgov/jinja2/v1/_layouts/base.html#L236-L285).
 
 #### Base.html
 
-This file serves as the base document for serving up assets and content.
-It's currently very complicated, obtrusive, and needs to be refactored.
+This file serves as the base HTML template for serving up assets and content. [View base.html on Github](https://github.com/cfpb/cfgov-refresh/blob/master/cfgov/jinja2/v1/_layouts/base.html).
```
Comment (Scotchester) [inline docs/atomic-structure.md]: This sentence could be rolled up into the section above. Having an entire section describing `base.html` seems unnecessary.

File: docs/notes-on-frameworks.md [hunk 6]
```diff
@@ -1,22 +0,0 @@
-# Notes on Frameworks
```
Comment (Scotchester) [inline docs/notes-on-frameworks.md]: Some of this content should serve as a baseline for a similar document on using frameworks that should live in the Development repo.

File: docs/atomic-structure.md (hunk 4 above)
Comment (sebworks) [reply docs/atomic-structure.md]: yep, they can be removed.

File: docs/atomic-structure.md (hunk 1 above)
Comment (jimmynotjim) [reply docs/atomic-structure.md]: We've never really followed the atomic template principals, but maybe we should to stay consistent?

Comment (anselmbradford) [reply docs/atomic-structure.md]: It was used for a time in `.t-careers`.

File: docs/atomic-structure.md (hunk 2 above)
Comment (cfarm) [reply docs/atomic-structure.md]: Done - added to Platform GHE

**Pull request merged**